language/stdlib/modules/libra_time.mvir

```diff
@@ -0,0 +1,28 @@
+module LibraTime {
+    struct T {
+        // A unix timestamp
+        timestamp: u64,
+    }
+
+    public new(timestamp: u64): Self.T {
+        return T {timestamp: move(timestamp)};
+    }
+
+    public get_timestamp(t: &Self.T): u64 {
+        return *(&move(t).timestamp);
+    }
+
+    public set_timestamp(t: &mut Self.T, new_time: u64) {
+        *(&mut move(t).timestamp) = move(new_time);
+        return;
+    }
+
+    public add_millis(t: &mut Self.T, elapsed: u64) {
+        *(&mut move(t).timestamp) =  *(&copy(t).timestamp) + move(elapsed);
+        return;
+    }
+
+    public is_before(lhs: &Self.T, rhs: &Self.T): bool {
```

Comment:
- Perhaps `is_before_exclusive` to make it explicit that this is exclusive of the `rhs`?
- A variant of this where `rhs` is a `u64` is probably useful as well.
- We would probably also want `is_after_exclusive` (or similar)